artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java

```diff
@@ -1623,6 +1623,32 @@ public void testAddressSettingsPageLimitInvalidConfiguration5() throws Throwable
       Assert.assertEquals(null, storeImpl.getPageFullMessagePolicy());
    }
 
+   @Test
+   public void testPagePrefetch() throws Throwable {
+      ConfigurationImpl configuration = new ConfigurationImpl();
+
+      Properties properties = new Properties();
+
+      String randomString = RandomUtil.randomString();
+
+      properties.put("addressSettings.#.expiryAddress", randomString);
+      properties.put("addressSettings.#.prefetchPageMessages", "333"); // this should make the PagingStore to parse it as null
+      properties.put("addressSettings.#.prefetchPageBytes", "777"); // this should make the PagingStore to parse it as null
+
+      configuration.parsePrefixedProperties(properties, null);
+
+      Assert.assertEquals(1, configuration.getAddressSettings().size());
+      Assert.assertEquals(SimpleString.toSimpleString(randomString), configuration.getAddressSettings().get("#").getExpiryAddress());
+      Assert.assertEquals(333, configuration.getAddressSettings().get("#").getPrefetchPageMessages());
+      Assert.assertEquals(777, configuration.getAddressSettings().get("#").getPrefetchPageBytes());
+
+      PagingStore storeImpl = new PagingStoreImpl(new SimpleString("Test"), (ScheduledExecutorService) null, 100L, Mockito.mock(PagingManager.class), Mockito.mock(StorageManager.class), Mockito.mock(SequentialFileFactory.class), Mockito.mock(PagingStoreFactory.class), new SimpleString("Test"), configuration.getAddressSettings().get("#"), null, null, true);
+
+      Assert.assertEquals(333, storeImpl.getPrefetchPageMessages());
+      Assert.assertEquals(777, storeImpl.getPrefetchPageBytes());
+   }
+
+
```

Comment:
A test [somewhere] checking the values are are as expected when _not_ explicitly set would seem appropriate.

Reply:
I need to do this one still... doing it now.